fuzzy-search.js

```diff
@@ -49,19 +51,17 @@ function fuzzySearch(haystackList, needle, renderer, caseSensitive = false) {
 
 		// Check the score of searchItem and act accordingly
 		if (score == 1) {
-			preciseMatch.push(haystackList[i]);
+			preciseMatch += `${haystackList[i]} `;
 		} else if (score == 2) {
-			preciseMatch.splice(0, 0, haystackList[i]);
+			preciseMatch = `${haystackList[i]} ${preciseMatch} `;
 		} else {
-			score = isFuzzyMatch(searchItem);
-			if (score) {
+			if (isFuzzyMatch(searchItem)) {
 				fuzzyMatch.push({ item: haystackList[i], score });
 			}
 		}
 	}
-
 	sortArray(fuzzyMatch);
-	return renderer(preciseMatch, fuzzyMatch);
+	return renderer(preciseMatch.split(' '), fuzzyMatch);
```

Comment:
`String.prototype.split()` is not that expensive.. Array of 10000 elements can be constructed from a string, in approximately  4.8 seconds, as shown in below screenshot
![image](https://user-images.githubusercontent.com/3054432/104835866-818e6700-58cf-11eb-82e7-c5b77a7a7420.png)